### DISCUSSION TITLE
Understanding BOINC Architecture

### DISCUSSION THREAD

**Discussion by amgfrthsp** [BOINC/boinc#5506] [Q&A] [answered]
Hi Volunteer Computing community! I am trying to dive deeper into how BOINC works from an SWE point of view. After studying wiki and relevant papers I came up with a couple of questions. I would be happy if you could help me with them.

Where jobs application, input and output files are stored? afaik workunit table holds only metadata for such files, and clients make requests to a data server instance in order to download necessary files and upload them. What happens to these files there? Where are they stored? Is there a separate database or file storage for that?

**Comment by davidpanderson** [comment, marked as answer]
See e.g. https://github.com/BOINC/boinc/wiki/Job-processing-(introduction)

**Reply by amgfrthsp** [reply]
Thanks a lot! Missed this one apparently
